packages/ui-react/src/containers/HeaderSearch/HeaderSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef } from 'react';
+import { useLocation } from 'react-router';
+import { useConfigStore } from '@jwp/ott-common/src/stores/ConfigStore';
+import { shallow } from '@jwp/ott-common/src/utils/compare';
+import { isTruthyCustomParamValue } from '@jwp/ott-common/src/utils/common';
+import { useUIStore } from '@jwp/ott-common/src/stores/UIStore';
+import SearchIcon from '@jwp/ott-theme/assets/icons/search.svg?react';
+import CloseIcon from '@jwp/ott-theme/assets/icons/close.svg?react';
+
+import Icon from '../../components/Icon/Icon';
+import SearchBar from '../../components/SearchBar/SearchBar';
+import useSearchQueryUpdater from '../../hooks/useSearchQueryUpdater';
+import HeaderActionButton from '../../components/Header/HeaderActionButton';
+
+import styles from './HeaderSearch.module.scss';
+
+const HeaderSearch = () => {
```

Comment:
Should we also make it part of the Header component?

Reply:
This also is a container (although this one also contains UI...). The UI can be split into a component in the Header folder, but the container remains. We probably want to make a better/stricter convention for this

Reply:
Okay I see your point now, using components for all stateless UI and containers for stateful. My concern with the header is that it is split into several folders now across components and containers.

Maybe `features` folder could help us in the future where we would group some connected containers/components (Header (or Layout for Cards, Header and Footer), Form, EPG, Payments, Subscriptions ...)? Components could be just for something we reuse across features (maybe also stateful). Just thinking out loud now.

Reply:
Thanks, Anton! Yes, I remember @dbudzins also suggesting something similar. Let's discuss this in the near future! 😄